defaults/main.yml

```
@@ -1,6 +1,8 @@
 ---
 # defaults file for galaxyproject.cvmfs
 
+cvmfs_apt_repository_url: https://cvmrepo.s3.cern.ch/cvmrepo/apt/
```

Could you use `cvmfs_apt_repository` instead. This is the format we use in most of our other roles.
Also, it may be better to define the whole deb line so we could tweak it further if required.
```suggestion
cvmfs_apt_repository: "deb [allow-insecure=true] https://cvmrepo.s3.cern.ch/cvmrepo/apt/ {{ ansible_distribution_release }}-prod main"
```

---

Defining the whole deb line won't work with the 3rd task at line 32.
`repo: deb [allow-insecure=true] {{ cvmfs_apt_repository_url }} xenial-prod main`

---

or I can change the repo for the first 2 tasks, and ignore the 3rd one and keep it hard-coded to cvmrepo.s3.cern.ch...?

---

This is legacy code for Ubuntu releases older than Xenial (16.04). It could have been useful with static URLs, but it seems irrelevant now. I guess you could simply remove this task from the role.

I also wonder if the task at line 19 is still relevant. I guess you could remove the `when: ansible_distribution != 'Ubuntu'` in the first task and let the user define something different from the default if required.

Am I missing something?